### PR TITLE
Fix/201 detect file change

### PR DESCRIPTION
**Pull request by ConnorStroomberg**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [NA] User documentation updated
- [x] Clean commits
- [x] No warnings during install
- [NA] Updated flow typing

### CODE REVIEW

**Comment by fdlk** [comment]
I can see the classes get updated. I'd expect some kind of clear button to be present because the field is not required.

File: test/e2e/specs/file-field-test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    browser.end()
+  },
+
+  'File field allow selecting a file': function (browser) {
+    // Skip set file test for safari as safari does not allow to post a file
+    if (browser.options.desiredCapabilities.browserName !== 'safari') {
+      browser.options.desiredCapabilities.name = 'File field allow selecting a file'
+      browser.expect.element('#alt-file-example').to.be.present
+      // bootstrap uses label to fake file input value
+      browser.uploadFile(path.resolve(path.join(__dirname, 'file-field-test.js')), '#alt-file-example')
```
Comment (fdlk) [inline test/e2e/specs/file-field-test.js]: Nice!

File: src/components/field-types/FileFieldComponent.vue [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
+      </label>
       <div class="custom-file">
         <label class="custom-file-label" :for="field.id">
           {{ label }}
         </label>
 
         <input
-          :id="field.id"
-          :name="field.id"
+          :id="'alt-' + field.id"
```
Comment (fdlk) [inline src/components/field-types/FileFieldComponent.vue]: There may be an attribute with that name, e.g. an entity type with attributes file and alt-file.
I think then the element ids will collide.

**Comment by fdlk** [comment]
If the file is not updated by the user, the submit event will contain the file name string. If it does get updated, the event contains a File object. Is it the responsibility of the caller to detect the change and make sure things are handled properly depending on what happened?

**Comment by ConnorStroomberg** [comment]
Yes, not only does the event contain the File object, so does the fromData object. It is up to the user of the ui-form component to handle creating/updating/... .  It can use events to get notified, but most likely wil use the fromData object for submitting the data to a server ( it that is what is needed)

**Comment by ConnorStroomberg** [comment]
Created a separate issue for adding a clear button:

https://github.com/molgenis/molgenis-ui-form/issues/215

**Comment by fdlk** [comment]
Current forms have a somewhat dubiously labeled "change" button which clears the value.

**Pull request merged**